mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -1253,18 +1252,116 @@ impl TendermintCoin {
         sign_doc.sign(&signkey)
     }
 
-    pub(super) fn any_to_sign_doc(
+    pub(super) fn any_to_serialized_sign_doc(
         &self,
         account_info: BaseAccount,
         tx_payload: Any,
         fee: Fee,
         timeout_height: u64,
         memo: String,
-    ) -> cosmrs::Result<SignDoc> {
+    ) -> cosmrs::Result<SerializedUnsignedTx> {
         let tx_body = tx::Body::new(vec![tx_payload], memo, timeout_height as u32);
         let pubkey = self.activation_policy.public_key()?.into();
         let auth_info = SignerInfo::single_direct(Some(pubkey), account_info.sequence).auth_info(fee);
-        SignDoc::new(&tx_body, &auth_info, &self.chain_id, account_info.account_number)
+        let sign_doc = SignDoc::new(&tx_body, &auth_info, &self.chain_id, account_info.account_number)?;
+
+        let tx_json = json!({
+            "sign_doc": {
+                "body_bytes": sign_doc.body_bytes,
+                "auth_info_bytes": sign_doc.auth_info_bytes,
+                "chain_id": sign_doc.chain_id,
+                "account_number": sign_doc.account_number,
+            }
+        });
+
+        Ok(SerializedUnsignedTx {
+            tx_json,
+            body_bytes: sign_doc.body_bytes,
+        })
+    }
+
+    /// This should only be used for Keplr from Ledger!
+    /// When using Keplr from Ledger, they don't accept `SING_MODE_DIRECT` transactions.
+    ///
+    /// Visit https://docs.cosmos.network/main/build/architecture/adr-050-sign-mode-textual#context for more context.
+    pub(super) fn any_to_legacy_amino_json(
+        &self,
+        account_info: BaseAccount,
+        tx_payload: Any,
+        fee: Fee,
+        timeout_height: u64,
+        memo: String,
+    ) -> cosmrs::Result<SerializedUnsignedTx> {
+        const MSG_SEND_TYPE_URL: &str = "/cosmos.bank.v1beta1.MsgSend";
+        const LEDGER_MSG_SEND_TYPE_URL: &str = "cosmos-sdk/MsgSend";
+
+        // Ledger's keplr works as wallet-only, so `MsgSend` support is enough for now.
```

Comment:
I think the below `wallet_only` method should be implemented for tendermint coin and token the same way while adding a case for returning `true` if `is_keplr_from_ledger` is `true`
https://github.com/KomodoPlatform/komodo-defi-framework/blob/fa71adb8aadffa7eb38608ea0b6b14893e779d53/mm2src/coins/lp_coins.rs#L3194-L3198

Reply:
Done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2148/commits/11c0a8974421effd65c1d26415db00bf6cee7f28